### DISCUSSION TITLE
Anonymous volume inside a mounted volume

### DISCUSSION THREAD

**Discussion by ghost** [containers/podman#18593] [Q&A] [answered]
Is it possible to make an anonymous volume inside of a mounted volume?

My use-case is that i want to setup a js development environment. My choice is to mount the project directory and do all the stuff inside it. Node would obviously create a "node_modules" folder inside the project directory, which I don't like leaking into my host environment.

So far I thought I'd create an anonymous volume mounted at that "node_modules" path, but it doesn't seem to work. I do it like this (using Makefile):
```Makefile
PWD = $(shell pwd)
DIR = $(shell basename $(PWD))
MNT = /opt/$(DIR)
IMG = node

sh:
	podman run -v "$(PWD):$(MNT):z" -v "$(MNT)/node_modules" -p "9500:9500" -w $(MNT) -it $(IMG) bash
```

So am I doing this right? Is it just not a viable way to use volumes? Maybe there's some other way to isolate node_modules that I don't know of? Please kindly provide your suggestions, if it wouldn't be terribly inconvenient.

**Comment by Luap99** [comment]
This should just work:
```
podman run -v ~/test:/test:Z -v /test/subdir --rm alpine mount | grep test
/dev/mapper/luks-7c7d65ef-20b6-462f-8b6d-29403fc74ba9 on /test type btrfs (rw,...)
/dev/mapper/luks-7c7d65ef-20b6-462f-8b6d-29403fc74ba9 on /test/subdir type btrfs (rw,...)

```

**Reply by ghost** [reply]
That actually does work but files are missing from the directory, it seems as though it creates a new volume every time. I think I'll try making a named volume.

**Reply by ghost** [reply]
Oh wait, I don't think I can make it not anonymous. Can I make a named volume without having it mounted to my host?

**Reply by Luap99** [reply]
Yes that is the point of an anonymous volume, it is creates new one for the container and is deleted with the container.
If you want them to persist a named volume is the way to go.

**Reply by Luap99** [reply, marked as answer]
Named or anonymous the directory will always exists on the host. It is just a bind mount from the podman volume directory into the container.

**Reply by ghost** [reply]
OK, that makes sense, thank you very much!
